### PR TITLE
Add an helper to check if a slot has been filled in

### LINKED ISSUE #206: Helper to know if a slot has been filled
In my component I have a wrapper around an optional slot.

```html
<div class="border-b border-gray-200 px-4 py-4 flex items-center justify-between sm:px-6 lg:px-8">
  <div class="flex-1 min-w-0 py-2">
    <h1 class="text-lg font-medium leading-6 text-gray-900 sm:truncate">
      {{ @page_title }}
    </h1>
  </div>
  <div class="flex">
    <slot name="right"/>
  </div>
</div>
```
As the slot is optional, if the user does not fill it, the result contains an empty `div`:
```html
<div class="flex">
</div>
```

I would like to remove this empty `div` by knowing if the slot has been filled. Today I created a helper for this:
```elixir
  defp has_slot?(%{__surface__: %{provided_templates: slots}}, slot) do
    slot in slots
  end

  defp has_slot?(_, _), do: false
```
And I use it like that
```html
<div class="border-b border-gray-200 px-4 py-4 flex items-center justify-between sm:px-6 lg:px-8">
  <div class="flex-1 min-w-0 py-2">
    <h1 class="text-lg font-medium leading-6 text-gray-900 sm:truncate">
      {{ @page_title }}
    </h1>
  </div>
  <div class="flex" :if={{ has_slot?(assigns, :right) }}>
    <slot name="right"/>
  </div>
</div>
```

Does Surface have a more idiomatic way to know is a slot has already been filled? If not, what about adding a helper in Surface?

I would be happy to implement this feature if you wish.

### PR DESCRIPTION
**Pull request by Malian**
This pull request fixes #206 

Now you can use a macro inside templates to check if a slot has been filled in or not.

```html
    <div :if={{ slot_assigned?(:header) }}>
      <slot name="header"/>
    </div>
```
If the slot does not exist, a warning is generated to help the user to fix the issue.

### CODE REVIEW

**Comment by Malian** [comment]
@msaraiva &  @lnr0626 I am not sure I use the right way to test the behaviour. I managed the `default` slot edge case and the "pretty" warning message. About the warning, I am not fan as the function is 95% the same as the one in the compiler.ex file. I am open to any suggestion to improve this pull request.

File: test/slot_test.exs [hunk 1]
```diff
@@ -117,6 +117,34 @@ defmodule Surface.SlotTest do
...
+    use Surface.Component
+
+    slot default
+    slot header
+    slot footer
+
+    def render(assigns) do
+      ~H"""
+      <div>
+        <header :if={{ slot_assigned?(:header)}}>
```
Comment (msaraiva) [inline test/slot_test.exs:130]: Just to keep it symmetric :)

```suggestion
        <header :if={{ slot_assigned?(:header) }}>
```

File: test/slot_test.exs [hunk 2]
```diff
@@ -117,6 +117,34 @@ defmodule Surface.SlotTest do
...
+          <slot name="header"/>
+        </header>
+        <main :if={{ slot_assigned?(:default)}}>
```
Comment (msaraiva) [inline test/slot_test.exs:133]:
```suggestion
        <main :if={{ slot_assigned?(:default) }}>
```

**Review by msaraiva — COMMENTED** [review]
Hey @Malian!

Great work! I believe the only thing missing is a test for the warning message. Please, let me know if you need any assistance on that regard.

Cheers.

**Comment by Malian** [comment]
@msaraiva Thanks for your feedbacks.

I added a test to catch the warning message but it does not pass.

This is the expected output:
```
           no slot "heade" defined in the component 'Elixir.Surface.SlotSyncTest.TestComponentWithWrongOptionalSlotName'

             Did you mean "header"\?

             Available slots: "default", "footer" and "header"
             code:10:\
```
but I receive the following one with the `[warning]`, the escaped spaces and double quotes:
```\e[33mwarning: \e[0mno slot \"heade\" defined in the component 'Elixir.Surface.SlotSyncTest.TestComponentWithWrongOptionalSlotName'\n\n  Did you mean \"header\"?\n\n  Available slots: \"default\", \"footer\" and \"header\"\n  code.exs:11: Surface.SlotSyncTest.TestComponentWithWrongOptionalSlotName.render/1\n\n```

It's probably due to the code I use to extract the warning message that I copied from `compiler_test.ex`
```elixir
  output =
      capture_io(:standard_error, fn ->
        {{:module, _, _, _}, _} =
          Code.eval_string(component_code, [], %{__ENV__ | file: "code.exs", line: 1})
      end)
```

Do you have any idea how to fix this?

File: test/slot_test.exs [hunk 3]
```diff
@@ -700,4 +795,45 @@ defmodule Surface.SlotSyncTest do
...
+          Code.eval_string(component_code, [], %{__ENV__ | file: "code.exs", line: 1})
+      end)
+
+    assert output =~ ~r"""
+           no slot "heade" defined in the component 'Elixir.Surface.SlotSyncTest.TestComponentWithWrongOptionalSlotName'
+
+             Did you mean "header"\?
+
+             Available slots: "default", "footer" and "header"
+             code:10:\
```
Comment (msaraiva) [inline test/slot_test.exs:836]: I believe this is all you need:

```suggestion
             code.exs:11\
```

Comment (Malian) [reply test/slot_test.exs]: Good catch. Thank you.

**Comment by msaraiva** [comment]
@Malian thanks! ❤️
